spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -244,10 +248,25 @@ public SparkTable alterTable(Identifier ident, TableChange... changes) throws No
 
   @Override
   public boolean dropTable(Identifier ident) {
+    return dropTableInternal(ident, false);
+  }
+
+  @Override
+  public boolean purgeTable(Identifier ident) {
+    return dropTableInternal(ident, true);
+  }
+
+  private boolean dropTableInternal(Identifier ident, boolean purge) {
     try {
+      if (purge) {
+        Table table = load(ident).first();
+        ValidationException.check(
+            PropertyUtil.propertyAsBoolean(table.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
+            "Cannot purge table: GC is disabled (deleting files may corrupt other tables)");
+      }
       return isPathIdentifier(ident) ?
```

Comment:
I know we used a ternary operator here before but the whole statement seems too complicated. I'd consider refactoring this into a separate utility method like `dropTableWithoutPurging` in the example above.